scripts/irods/test/test_itree.py

```diff
@@ -46,6 +46,23 @@ def test_depth(self):
         for index,item in enumerate(directory_names):
             self.admin.assert_icommand(["itree", "-L", str(1 + index)], 'STDOUT_SINGLELINE', ["  " * (1 + index) + item])
 
+    def test_pattern_and_ignore_options_together__6791_6806(self):
+        patterns = ['foo', 'bar', 'bat']
+        for i in range(10):
+            for p in patterns:
+                self.admin.assert_icommand(['itouch', p + str(i)])
+
+        # Using forward and reverse combination of -P and -I options, catch objects matching /bar[0-9]/
```

Comment:
matches `/bat[0-9]/`, rather than `bar`, right?

Reply:
The `-I` option has a value of `*ar?`, which would suggest that `bar` would match and `bat` would not,?